### PR TITLE
Add Brands Table

### LINKED ISSUE #46: Add Brands table
## REQUIREMENTS / DESCRIPTION

- Add a brand table to the DB
- Add brand CRUD
- Modify products created to use registered brands
- Homepage should show Brands banner
- Bellow the banner products with that banner's brand should be displayed

### PR DESCRIPTION
**Pull request by Ballo1988**
# Description

What did you implemented/Why did you implemented this:

- Added an API from Google that allows users to select their location and set their address in a more intiutive way
- all Address fields, but phone number, now autocompletes itself when selecting a location from the map
- Added Brands table, now admin can register brands and attach many images to it
- Modified Products, so now admin has to select a brand from already registered brand
- Modified Products, so they now can have many images.
- Modified Products form so now is more user friendly (now total price calculates itself!)
- Added some validations to prevent costumers from getting into trouble
random user emails and names.

## Screenshots
image
image
image
image
image
image
image
image
image

Closes #46

### CODE REVIEW

File: app/views/addresses/_form.html.erb [hunk 1]
```diff
@@ -33,4 +37,57 @@
...
+
+      // Extraer la parte de la dirección correspondiente a la calle, número y colonia
+      const streetNumber = place.address_components.find(component => component.types.includes("street_number"));
+      const route = place.address_components.find(component => component.types.includes("route"));
+      const neighborhood = place.address_components.find(component => component.types.includes("neighborhood"));
+      const addressValue = `${route ? route.long_name : ''}${streetNumber ? ' ' + streetNumber.long_name : ''}${neighborhood ? ', ' + neighborhood.long_name : ''}`;
+      document.getElementById("address-input").value = addressValue;
+    }
+  }
+</script>
```
Comment (juperameza) [inline app/views/addresses/_form.html.erb:91]: You shouldnt do javascript like this. Use stimulus js controllers and add them to an element, you can also pass data back and forward. Heres a [video example](https://www.youtube.com/watch?v=nf88pbG1a6s) and [another one ](https://www.youtube.com/watch?v=pjYwjOShp0c) and a [blog ](https://dev.to/briankephart/passing-arguments-to-stimulusjs-controllers-1kgi)

**Review by juperameza — CHANGES_REQUESTED** [review]

**Review by juperameza — APPROVED** [review]
lgtm

File: test/models/brand_test.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class BrandTest < ActiveSupport::TestCase
+  # test "the truth" do
+  #   assert true
+  # end
```
Comment (PudinDeveloper) [inline test/models/brand_test.rb:6]: Nothing important, just remove this extra lines.

**Review by PudinDeveloper — APPROVED** [review]
LGTM

**Pull request merged**